internal/provider/resource_vault_cluster.go

```diff
@@ -56,6 +56,13 @@ func resourceVaultCluster() *schema.Resource {
 				ForceNew:         true,
 				ValidateDiagFunc: validateSlugID,
 			},
+			"tier": {
+				Description:      "Tier of the HCP Vault cluster. Valid options for tiers - `development`, `small`, `medium`, `large`",
+				Type:             schema.TypeString,
+				Required:         true,
```

Comment:
In order to avoid forcing existing Vault clusters to be recreated since they don't have a `tier` input, you can mark this as `Computed: true`. That way any clusters without `tier` can just rely on the default set by the API, `development`.

Reply:
Based on [this](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html) then it cannot be required:

```
Required
Data structure: bool
Values: true Restrictions:
Cannot be used if Optional is true
Cannot be used if Computed is true
Must be set if Optional is omitted and element is not Computed
```

Reply:
Oh yes you're right. Let's go with Optional. I'll see if we can change that on the Consul cluster for consistency.

Reply:
Is there a difference between `Required: false` and `Optional: true`?

Reply:
Indeed there is :-)

```
=== RUN   TestAccVaultCluster
    resource_vault_cluster_test.go:37: Step 1/4 error: Error running pre-apply refresh: exit status 1
        
        Error: Computed attributes cannot be set
        
          on terraform_plugin_test.tf line 13, in resource "hcp_vault_cluster" "test":
          13: 	tier									= "standard_small"
        
        Computed attributes cannot be set, but a value was set for "tier".
--- FAIL: TestAccVaultCluster (0.47s)
```